frontend/src/concepts/modelCatalog/context/ModelCatalogContext.tsx

```diff
@@ -1,29 +1,32 @@
 import React from 'react';
+import { FetchState } from '~/utilities/useFetchState';
 import { ModelCatalogSource } from '~/concepts/modelCatalog/types';
-import { modelCatalogSourceMock } from '~/concepts/modelCatalog/mockData/modelCatalogSourceMock';
+import { useModelCatalogSources } from '~/concepts/modelCatalog/useModelCatalogSources';
 
 export type ModelCatalogContextType = {
-  modelCatalogSources: ModelCatalogSource[];
+  modelCatalogSources: FetchState<ModelCatalogSource[]>;
 };
 
 type ModelCatalogContextProviderProps = {
   children: React.ReactNode;
 };
 
 export const ModelCatalogContext = React.createContext<ModelCatalogContextType>({
-  modelCatalogSources: [],
+  modelCatalogSources: [[], false, undefined, () => Promise.resolve(undefined)],
 });
 
 export const ModelCatalogContextProvider: React.FC<ModelCatalogContextProviderProps> = ({
   children,
 }) => {
-  const modelCatalogSourcesMock = React.useMemo(() => [modelCatalogSourceMock({})], []);
+  const modelCatalogSources = useModelCatalogSources();
```

Comment:
You can wrap this in `useMakeFetchObject`. This changes the array return value to an object with named properties, easier to deal with.

```ts
const modelCatalogSources = useMakeFetchObject(useModelCatalogSources());
```